lessons/specifics/iex_helpers.md

```diff
@@ -0,0 +1,250 @@
+
+### `i`
+Since we already have another tool to learn about this and its valid Elixir code, why not use `h` to learn about `i`? Here we go:
+
+```elixir
+iex> h i
+
+                                  def i(term)
+
+Prints information about the given data type.
+
+iex> i Map
+Term
+  Map
+Data type
+  Atom
+Module bytecode
+  /usr/local/Cellar/elixir/1.3.3/bin/../lib/elixir/ebin/Elixir.Map.beam
+Source
+  /private/tmp/elixir-20160918-33925-1ki46ng/elixir-1.3.3/lib/elixir/lib/map.ex
+Version
+  [9651177287794427227743899018880159024]
+Compile time
+  no value found
+Compile options
+  [:debug_info]
+Description
+  Use h(Map) to access its documentation.
+  Call Map.module_info() to access metadata.
+Raw representation
+  :"Elixir.Map"
+Reference modules
+  Module, Atom
+```
+
+So now we have a bunch more information about `Map`, including all the way down to where its source is stored and the modules it references.
+This is quite useful when exploring custom and foreign data types and new functions.
+The individual headings can be dense, but at a high level we can gather some relevent information:
+
+- Its an atom data type
+- Where the source code is
+- The version, and compile options
+- A general description
+- How to access it
+- What other modules it references
+
+This is a lot to work with, and much better than going in blind.
+
+### `r`
+`r` is our helper for recompiling a given module.
+Say you have changed some code and simply want to run the new function you added, you could save the file then run:
+
+```elixir
+iex> r MyProject
+warning: redefining module MyProject (current version loaded from _build/dev/lib/my_project/ebin/Elixir.MyProject.beam)
+  lib/my_project.ex:1
+
+{:reloaded, MyProject, [MyProject]}
+```
+
+### `s`
+`s` gives us type spec information for a given module or function.
+This way we can know what it is expecting:
+
+```elixir
+iex> s Map.merge/2
+@spec merge(map(), map()) :: map()
+
+# it also works on entire modules
+iex> s Map
+@spec get(map(), key(), value()) :: value()
+@spec put(map(), key(), value()) :: map()
+# ...
+@spec get(map(), key()) :: value()
+@spec get_and_update!(map(), key(), (value() -> {get, value()})) :: {get, map()} | no_return() when get: term()
+```
+
+### `t`
+`t` Tells us about Types available in a given module:
+
+```elixir
+iex> t Map
+@type key() :: any()
+@type value() :: any()
+```
+
+And now we know that Map defines key and value types in its implementation.
+If we go and look at the source of `Map`:
+
+```elixir
+defmodule Map do
+# ...
+  @type key :: any
+  @type value :: any
+# ...
+```
+
+This is a simple example, stating that keys and values per the implementation can be any type, but it is useful to know.
+
+## Conclusion
+By leveraging all these built-in niceties, it can become a lot easier to explore and write new code. IEx is very robust, and does a lot to empower developers. With these tools in your toolbox, exploring and building should be even more fun!
```

Comment:
By leveraging all these built-in niceties we can easily explore the code and learn more about how things work.  IEx is very powerful and robust tool that empower developers.  With these tools in our toolbox, exploring, and building can be even more fun!